src/poetry/console/commands/search.py

```diff
@@ -21,11 +21,14 @@ class SearchCommand(Command):
     ]
 
     def handle(self) -> int:
-        from poetry.repositories.pypi_repository import PyPiRepository
+        seen = set()
 
-        results = PyPiRepository().search(self.argument("tokens"))
+        for result in self.poetry.pool.search(self.argument("tokens")):
+            if result.pretty_string in seen:
+                continue
+
+            seen.add(result.pretty_string)
 
-        for result in results:
             self.line("")
```

Comment:
not sure if this is necessary, it makes "holes" in the list of packages.

Reply:
This might be out of scope for this change, it is already bigger than I originally anticipated. I can raise another one improving the output formatting and adding repository selection.

The spacing is required because originally we also add the description. We might need a better (show like) output for this one. But it is another change I think.

Reply:
Okay, I improved the output too now. 

```console
 Package             Version Source  Description 
 pytorch-triton      2.3.0   pytorch             
 pytorch-triton      3.0.0   pytorch             
 pytorch-triton-rocm 2.0.0   pytorch             
 pytorch-triton-rocm 2.0.1   pytorch             
 pytorch-triton-rocm 2.1.0   pytorch             
 pytorch-triton-rocm 2.2.0   pytorch             
 pytorch-triton-rocm 2.3.0   pytorch             
 pytorch-triton-rocm 2.3.1   pytorch             
 pytorch-triton-rocm 3.0.0   pytorch             
 pytorch-triton-rocm 3.1.0   pytorch  
```